### PR TITLE
OSDOCS#11133: Add a link for async RN for HCP

### PR DESCRIPTION
**Pull request by xenolinux**
This PR removes the HCP RN (bug fixes and 4.17 entry from the TP table) in favor of [newly added HCP RN for its async releases](https://github.com/openshift/openshift-docs/pull/83594). 

Version(s): 4.17

Issue: https://issues.redhat.com/browse/OSDOCS-11133

- Link to docs preview: https://84105--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-17-release-notes.html#ocp-4-17-bug-fixes_release-notes -- bug fixes section
- https://84105--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-17-release-notes.html#ocp-4-17-technology-preview-tables_release-notes -- TP tracker table and the note above it

QE review:
- [x] QE has approved this change.

Additional information: The PR contains the change management approvals

### CODE REVIEW

**Comment by zanetworker** [comment]
lgtm

**Comment by xltian** [comment]
LGTM

**Review by lahinson — APPROVED** [review]
@xenolinux Thanks so much for making this update! I posted a couple of comments for your consideration :)

**Comment by mwerner2113** [comment]
LGTM

File: release_notes/ocp-4-17-release-notes.adoc [hunk 1]
```diff
@@ -2609,52 +2579,49 @@ In the following tables, features are marked with the following statuses:
...
-|Technology Preview
 
 |{hcp-capital} for {product-title} on {rh-openstack}
 |Not Available
 |Not Available
-|Developer Preview
 
 |====
 
+[NOTE]
```
Comment (kalexand-rh) [inline release_notes/ocp-4-17-release-notes.adoc:2620]: I think this note should go above the table. Otherwise, the change LGTM.

**Review by abhatt-rh — COMMENTED** [review]
LGTM!

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Pull request merged**